src/udev/udev-builtin-blkid.c

```diff
@@ -352,6 +420,29 @@ static int builtin_blkid(sd_device *dev, sd_netlink **rtnl, int argc, char *argv
         if (is_gpt)
                 find_gpt_root(dev, pr, test);
 
+        r = read_loopback_backing_inode(
+                        dev,
+                        fd,
+                        &backing_devno,
+                        &backing_inode,
+                        &backing_fname);
+        if (r < 0)
+                log_device_debug_errno(dev, r, "Failed to read loopback backing inode, ignoring: %m");
+        else if (r > 0) {
+                udev_builtin_add_propertyf(dev, test, "ID_LOOP_BACKING_DEVICE", DEVNUM_FORMAT_STR, DEVNUM_FORMAT_VAL(backing_devno));
+                udev_builtin_add_propertyf(dev, test, "ID_LOOP_BACKING_INODE", "%" PRIu64, (uint64_t) backing_inode);
+
+                if (backing_fname) {
+                        size_t l = strlen(backing_fname) * 4 + 1;
+                        char encoded[l];
```

Comment:
isn't this a VLA?

Reply:
Yes, but a short-lived and bounded one.

Reply:
By what is it bound?

Reply:
can you add an assert then? so that we document and enforce that